src/index/Index.cpp

```diff
@@ -290,16 +297,17 @@ void Index::createPermutationImpl(const string& fileName,
 
   LOG(INFO) << "Writing Meta data to file...\n";
 
-  metaData.appendToFile(&out);
+  //  metaData.appendToFile(&out);
   out.close();
   LOG(INFO) << "Permutation done.\n";
+  return std::move(metaData);
 }
 
 // ________________________________________________________________________
 template <class MetaData, class Comparator>
-void Index::createPermutation(ExtVec* vec,
-                              Permutation::PermutationImpl<Comparator> p,
-                              bool performUnique) {
+std::optional<MetaData> Index::createPermutation(
```

Comment:
Please add a method comment and make sure to mention in which cases the `std::optional` may be empty